### PR TITLE
Add users model

### PR DESCRIPTION
**Pull request by djosborne**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,3 +73,18 @@ The state table is very straightforward:
 ```
 
 There should be an index on the `device_id` column to enable quick searching.
+
+### `users`
+
+```
++-----------------------+
+| states                |
```
Comment (djosborne) [inline README.md]: Whoops. This should be ```users```

File: db/migrate/20150129032442_create_users.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateUsers < ActiveRecord::Migration
+  def change
+  	create_table :users do |t|
+  		t.string :username, 	 null: false
```
Comment (nahiluhmot) [inline db/migrate/20150129032442_create_users.rb]: Tabs < Spaces

File: app/cappy/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+require 'bcrypt'
+
+module Cappy
+  module Models
+    # This model represents the state of a connected device
+    class User < ActiveRecord::Base
+      self.table_name = 'users'
+
+      validates :username,      presence: true
```
Comment (nahiluhmot) [inline app/cappy/models/user.rb]: I'm not big on the extra spaces, @tlunter want to weigh in here?

File: app/cappy/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      validates :password_hash, presence: true
+
+      def password
+        @password ||= BCrypt::Password.new(password_hash)
+      end
+
+      def password=(new_password)
+        if new_password.nil?
+          # BCrypt::Password constructor creates a random hash if secret is nil.
+          # We want to reset the password if no password is passed in.
```
Comment (nahiluhmot) [inline app/cappy/models/user.rb]: Same here.

File: db/schema.rb [hunk 5]
```diff
@@ -34,4 +34,13 @@
...
   add_index "states", ["device_id"], name: "index_states_on_device_id"
 
+  create_table "users", force: :cascade do |t|
+    t.string   "username",      null: false
+    t.string   "password_hash", null: false
+    t.datetime "created_at",    null: false
+    t.datetime "updated_at",    null: false
+  end
+
+  add_index "users", ["username"], name: "index_users_on_username"
```
Comment (nahiluhmot) [inline db/schema.rb]: This should be a unique index.

File: db/migrate/20150129032442_create_users.rb (hunk 2 above)
Comment (tlunter) [inline db/migrate/20150129032442_create_users.rb]: ^^^ We should remove all hard tabs

File: db/migrate/20150129032442_create_users.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+  		t.string :password_hash, null: false
+  		t.timestamps 			 null: false
+  		
+  		t.index :username
```
Comment (tlunter) [inline db/migrate/20150129032442_create_users.rb]: This should be set to unique somehow.

File: db/schema.rb (hunk 5 above)
Comment (tlunter) [inline db/schema.rb]: Don't edit `schema.rb`, update the migration, rollback and retry it.

File: README.md [hunk 7]
```diff
@@ -73,3 +73,18 @@ The state table is very straightforward:
...
++-----------------------+
+| id:int                |
+| username:string 		|
```
Comment (tlunter) [inline README.md]: Indenting is breaking pipe location.

File: README.md [hunk 8]
```diff
@@ -73,3 +73,18 @@ The state table is very straightforward:
...
+
+### `users`
+
+```
++-----------------------+
+| users                 |
++-----------------------+
+| id:int                |
+| username:string       |
+| password:string       |
```
Comment (nahiluhmot) [inline README.md]: Could you update this to `password_hash`, don't want to imply that we're storing passwords in plain text :)

**Comment by nahiluhmot** [comment]
Perhaps we should be salting these passwords as well?

**Comment by nahiluhmot** [comment]
Closing this in favor of a cloud service.

**Comment by djosborne** [comment]
Why was this closed? I thought the MVP plan was to add Users?

**Comment by nahiluhmot** [comment]
@djosborne No, the MVP plan is to fake it in the UI. Reach goal (which I'll probably start next week) is a separate cloud service.

**Comment by djosborne** [comment]
When we spoke with Walleed we confirmed that we'd have a single set of working login credentials for our demo. I suppose we can fake it, even though that's not being entirely truthful. But I think closing this PR to make room for a  Nice-to-have is not the right move. I understand that any work done on logins will have to be removed when the cloud service sign on is implemented. But I think we should leave this open and in a state where it can be easily picked up, completed, and integrated, so that our MVP isn't completely relying on a stretch goal. That may just mean we never actually complete these features, but it's better than us getting screwed later on.

**Comment by nahiluhmot** [comment]
I'm working on the cloud service next week, so I feel confident that we can have a working prototype within a week or two.

**Pull request closed without merging**